packages/cardano-metamask-snap/src/api/sdk/transaction.ts

```diff
@@ -111,28 +123,41 @@ export const parseTransaction = ({
     .validityStartInterval()
     ?.toString();
 
-  const collateralReturn = parseCollateralReturn(
-    parsedTransactionBody.collateralReturn(),
-    ownAddresses,
-  );
-  const totalCollateral = parseTotalCollateral(
-    parsedTransactionBody.totalCollateral(),
-  );
-
-  const collateral =
-    collateralReturn || totalCollateral
-      ? {
-          collateralReturn,
-          totalCollateral,
-        }
-      : undefined;
+  if (hasPlutusTxFields(parsedTransactionBody)) {
```

Comment:
just wondering - is this logic mirroring the way we detect plutus transactions in trezor connect? If so, I'd consider linking that here for reference

Reply:
Yes, it follows how we in nufi determine if tx is a plutus one.
In nufi we reference ledger, we also do that here next to the definition of hasPlutusTxFields
https://github.com/cardano-foundation/ledgerjs-hw-app-cardano/blob/5e83052fd02c8e3bf88dbc837bb9219d80b41a40/src/parsing/transaction.ts#L341

Reply:
ok then I'd reference ledger here as well, for the sake of clarify